clippy_lints/src/methods.rs

```diff
@@ -1325,6 +1332,7 @@ fn lint_unnecessary_fold(cx: &LateContext<'_, '_>, expr: &hir::Expr, fold_args:
                     "this `.fold` can be written more succinctly using another method",
                     "try",
                     sugg,
+                    Applicability::HasPlaceholders,
```

Comment:
It may be good to change this to `Applicability::MaybeIncorrect`. See #3069 and #3303.

Reply:
Oh this is a stale PR to remind me (and others) that this should be done sometimes. When I'm (or someone else) continues working on this, I would start from scratch and go over these `Applicability` levels again (Also taking #3191 into consideration).

But yeah `Applicability::MaybeIncorrect` would probably be better here.